### ISSUE TITLE
Enable Link-Time Optimization (LTO)

### ISSUE THREAD

**Issue by zamazan4ik** [exaroth/liveboat#16]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a little bit. If you want to read more about LTO and its possible modes, I recommend starting from [this](https://doc.rust-lang.org/rustc/codegen-options/index.html#lto) Rustc documentation.

I think you can enable LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where in addition to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most optimized version of the application. However, if we enable it on the Cargo profile level for the Release profile, users, who install the application with `cargo install` will get the LTO-optimized version of the game "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48). You also could be interested in other optimization options like `codegen-units = 1`, etc.

Basically, it can be enabled with the following lines:

```
[profile.release]
codegen-units = 1
lto = true
```

I have made quick tests (Fedora 41, Rust 1.82, the latest version of the project) - here are the results:

* Release (current default): 16.5 Mib
* Release + `codegen-units = 1`: 14.7 Mib
* Release + Thin LTO: 16.6 Mib
* Release + Fat LTO: 14.1 Mib
* Release + `codegen-units = 1` + Thin LTO: 14.8 Mib
* Release + `codegen-units = 1` + Fat LTO: 13.7 Mib

Thank you. It would be nice if you would use LTO for your future Rust developments :)

**Comment by exaroth** [comment]
Hello!

Thank you very much for the input - I'm still fairly new to Rust and it's always great to find out some new tricks in the bag.

Did some local testing and indeed the binary size was significantly decreased after enabling the options you mentioned. Haven't tested performance increase but for this project in particular it's not really a priority.  Changes have been added [4ef48510332](https://github.com/exaroth/liveboat/commit/4ef48510332f15200322a1b5c0dddd7127523f36) and should be available to the users after next release

The binary size grew a lot after i added vendored dependency for sqlite and tls (wanted to avoid as many outside deps as possible) so any size decrease for the binary is a big win, as for the compilation time - it probably shouldn't have much of an effect on most users.

Cheers
